### PR TITLE
Prevent untrainable-batch stalls in RL training

### PR DESCRIPTION
**Pull request by wwwjn**
## Why

The Qwen3.5 debug integration test in https://github.com/pytorch/torchtitan/actions/runs/34312737649/job/102342893821 generated rollouts indefinitely without producing a training step. Random-init policies can give every rollout in a group the same reward. With `drop_zero_std_reward_groups=True`, every such group is discarded, so the generator keeps replenishing work while the trainer waits forever for a batch.

This is the same invariant addressed for earlier debug configs in #3802; the Qwen3.5 debug config added in #3957 omitted the override.

## What

- Set `drop_zero_std_reward_groups=False` for the Qwen3.5 debug config.
- Add the corresponding CLI override centrally to every command produced by the RL CI integration-test launcher, including multi-run tests.
- Record consecutive untrainable groups at the batcher boundary:
  - each complete batch-sized block of untrainable groups emits a warning;
  - any trainable group resets the count;
  - 10 consecutive untrainable batches raise `RuntimeError` instead of hanging indefinitely.
- Remove the resolved silent-hang TODO and add regression tests for the CI override, warnings, reset behavior, and terminal error.

This follows Prime-RL’s generalized no-progress protection: https://github.com/PrimeIntellect-ai/prime-rl/blob/main/src/prime_rl/orchestrator/train_sink.py

## Testing

- `python -m pytest torchtitan/experiments/rl/tests/test_async_controller.py torchtitan/experiments/rl/tests/test_rollout_utils.py -q` (27 passed)
- Targeted watchdog tests after the final naming change (3 passed).
- Targeted pre-commit hooks passed (Pyrefly skipped because RL experiments/tests are excluded by repository configuration and the configured local PyTorch tree is unavailable).
- Python compilation passed for all changed files.
- Validated that all 8 commands currently produced by the RL CI launcher contain the override.

The focused launcher pytest could not be collected locally because this checkout environment lacks `grain`; CI installs the project requirements.

### CODE REVIEW

File: torchtitan/experiments/rl/tests/integration_tests.py [hunk 1]
```diff
@@ -252,6 +257,16 @@ def build_rl_test_list() -> list[OverrideDefinitions]:
...
 
+    # CI can use random-init policies whose rollout groups all receive the same
+    # reward. Keep those groups so the trainer cannot wait forever for a batch.
+    for test in test_list:
+        test.override_args = [
+            [*override_args, _KEEP_ZERO_STD_REWARD_GROUPS]
+            for override_args in test.override_args
+        ]
+
+    return test_list
```
Comment (tianyu-l) [inline torchtitan/experiments/rl/tests/integration_tests.py:275]: make a warning if it's actually changed? o/w could be hard to track

**Review by tianyu-l — APPROVED** [review]

**Review by shuhuayu — APPROVED** [review]
lgtm, we do not need to consider loss convergence of ci tests targeting model enablement.

File: torchtitan/experiments/rl/tests/integration_tests.py [hunk 2]
```diff
@@ -28,8 +28,13 @@
 from torchtitan.tools.logging import logger
 
 
+_KEEP_ZERO_STD_REWARD_GROUPS = (
+    "--async-loop.training-sample-builder.no-drop-zero-std-reward-groups"
```
Comment (felipemello1) [inline torchtitan/experiments/rl/tests/integration_tests.py:32]: is the "no" in front of it a way to set it to false?

**Review by felipemello1 — APPROVED** [review]

**Comment by felipemello1** [comment]
thank you!

File: torchtitan/experiments/rl/tests/integration_tests.py (hunk 2 above)
Comment (wwwjn) [reply torchtitan/experiments/rl/tests/integration_tests.py:32]: Yes, in tyro CLI, if you wanto set a boolean flag to be false, add a "no-" to the name
